### PR TITLE
RA-1430: Prevent XSS in pop-up message after patient registration

### PR DESCRIPTION
**Pull request by isears**
https://issues.openmrs.org/browse/RA-1430

### CODE REVIEW

**Comment by isears** [comment]
Long time no PR! This is a long overdue fix for an issue discussed in this thread: https://talk.openmrs.org/t/security-issues/12339

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java [hunk 1]
```diff
@@ -263,7 +263,22 @@ public FragmentActionResult submit(UiSessionContext sessionContext, @RequestPara
             }
         }
 
-        InfoErrorMessageUtil.flashInfoMessage(request.getSession(), ui.message("registrationapp.createdPatientMessage", patient.getPersonName()));
+        // Ensure name is HTML-safe to prevent XSS in info messages
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java]: Can you just a single encodeHtml call to the return value of `ui.message("registrationapp.createdPatientMessage", patient.getPersonName())`?

**Comment by wluyima** [comment]
Thanks for your pull request @isears , I have created sub task under [RA-451](https://issues.openmrs.org/browse/RA-452) and assigned it to you so that we link this work to it.

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java (hunk 1 above)
Comment (isears) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java]: That could work. The one drawback is that if the `ui.message()` function ever returns any kind of html-formatted text (e.g. `<b>` or `<span>` tags), that text will get html-encoded and will not work correctly.

Comment (wluyima) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java]: I see, then you need to get to do this;

`ui.message("registrationapp.createdPatientMessage", ui.encodeHtml(ui.format(patient.getPersonName())));`

The key thing is to leave the 'stringfying' of the person name object to the internals of the framework.

Comment (wluyima) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java]: This is even better:

ui.message("registrationapp.createdPatientMessage",ui.encodeHtml(ui.format(patient)));

Comment (isears) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java]: Ok that sounds good to me. I will update.

Thanks!

**Comment by wluyima** [comment]
I tried to reproduce this on https://qa-refapp.openmrs.org or demo.openmrs.org and I was unable to with the given name as `&lt;script&gt;alert('xss')&lt;/script&gt;` how can I reproduce it?

**Comment by isears** [comment]
That given name should work.

I just checked w/given name `XSS <script>alert(1)</script>` and was able to reproduce.

Note that the **only** time this XSS executes is directly after the patient is created and the user is redirected to the patient's page. It will not execute on subsequent visits to the patient's page.

**Comment by wluyima** [comment]
I see, now am able too, before I was trying it by editing

**Pull request merged**